src/main/kotlin/org/rust/lang/core/psi/RustTokenTypes.kt

```diff
@@ -0,0 +1,26 @@
+package org.rust.lang.core.psi
+
+import com.intellij.lang.ASTNode
+import com.intellij.lang.DefaultASTFactory
+import com.intellij.openapi.components.service
+import com.intellij.psi.impl.source.tree.LeafPsiElement
+import com.intellij.psi.tree.IElementType
+import com.intellij.psi.tree.ILeafElementType
+import org.rust.lang.RustLanguage
+import org.rust.lang.core.psi.impl.RustLiteralImpl
+
+private val defaultASTFactory by lazy { service<DefaultASTFactory>() }
+
+open class RustTokenType(debugName: String) : IElementType(debugName, RustLanguage), ILeafElementType {
+    override fun createLeafNode(leafText: CharSequence?): ASTNode = LeafPsiElement(this, leafText)
+}
+
+class RustCommentTokenType(debugName: String) : RustTokenType(debugName) {
+    override fun createLeafNode(leafText: CharSequence?): ASTNode = defaultASTFactory.createComment(this, leafText!!)
+}
+
+class RustKeywordTokenType(debugName: String) : RustTokenType(debugName)
+
+class RustLiteralTokenType(debugName: String) : RustTokenType(debugName) {
+    override fun createLeafNode(leafText: CharSequence?): ASTNode = RustLiteralImpl(this, leafText!!)
```

Comment:
Let's have three types: for numbers, for chars and for strings. I don't think we should make disctinction between different flavors of strings and chars. We may want to distinguish floats and integers though, but I don't think we should: is `92f31` is an invalid float or invalid integer?